functions.php

```diff
@@ -241,4 +241,43 @@ function ValidateImg(array $files): string {
     return '';   
 }
 
+function addUser(mysqli $link, array $registration): bool {
+    $sql = 'INSERT INTO user (email, password, name, contacts) VALUES (?, ?, ?, ?)';
+    $stmt = db_get_prepare_stmt($link, $sql, $registration);
+    return mysqli_stmt_execute($stmt);
+}
+
+function validateEmail(mysqli $link, array $registration): string {
+    $email = $registration['email'];
+    if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
+         return 'Некорректный e-mail';
+    }
+
+    $sql = "SELECT email FROM user WHERE email = '" . $email . "'";
+    $result = mysqli_query($link, $sql);
+    if ($result) {
+        if (!empty(mysqli_fetch_all($result, MYSQLI_ASSOC))) {
+            return 'Данный e-mail занят';
+        }
+        return '';
```

Comment:
Это можно сделать чуть проще
```suggestion
$result = mysqli_fetch_all($result, MYSQLI_ASSOC));
return !empty($result) ? return 'Данный e-mail занят' : '';
```